java-security-test/src/main/java/com/sap/cloud/security/test/SecurityTest.java

```diff
@@ -189,46 +188,29 @@ public String getApplicationServerUri() {
 	}
 
 	void startApplicationServer() throws Exception {
-		WebAppContext context = createWebAppContext();
-		ServletHandler servletHandler = createServletHandler(context);
-
-		applicationServletsByPath
-				.forEach((path, servletHolder) -> servletHandler.addServletWithMapping(servletHolder, path));
-		applicationServletFilters.forEach((filterHolder) -> servletHandler
-				.addFilterWithMapping(filterHolder, "/*", EnumSet.of(DispatcherType.REQUEST)));
-
-		servletHandler
-				.addFilterWithMapping(new FilterHolder(new SecurityFilter()), "/*", EnumSet.of(DispatcherType.REQUEST));
-
-		applicationServer = new Server(applicationServerOptions.getPort());
-		applicationServer.setHandler(context);
-		applicationServer.start();
-	}
-
-	ServletHandler createServletHandler(WebAppContext context) {
-		ConstraintSecurityHandler security = new ConstraintSecurityHandler();
+		// create SecurityHandler
 		JettyTokenAuthenticator authenticator = new JettyTokenAuthenticator(
 				applicationServerOptions.getTokenAuthenticator());
+		ConstraintSecurityHandler security = new ConstraintSecurityHandler();
 		security.setAuthenticator(authenticator);
 
-		ServletHandler servletHandler = new ServletHandler();
-		security.setHandler(servletHandler);
-		context.setServletHandler(servletHandler);
+		// Create a ServletContextHandler
+		ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
+		context.setContextPath("/");
 		context.setSecurityHandler(security);
+		context.setResourceBase("src/main/java/webapp");
 
-		return servletHandler;
-	}
+		// add Servlets
+		applicationServletsByPath
+				.forEach((path, servletHolder) -> context.addServlet(servletHolder, path));
 
-	WebAppContext createWebAppContext() {
-		WebAppContext context = new WebAppContext();
```

Comment:
`WebAppContext` is an extension to `ServletContextHandler` and supports servlets configured in web.xml or by annotations without a need to explicitly set them in SecurityTestExtension like here:
```java 
SecurityTestExtension.forService(XSUAA)
			.useApplicationServer();
			.addApplicationServlet(HelloJavaServlet.class, HelloJavaServlet.ENDPOINT)
```
The way how you overwrote that is a simpler way and apps that use web.xml for example would need to define each servlet like mentioned above, meaning we take a way some flexibility, therefore, I would suggest we stick with the `WebAppContext`

Reply:
I was able to fix the 501 Jetty problem using WebAppContext now.